code/src/components/app-input-with-icon/AppInputWithIcon.test.tsx

```diff
@@ -0,0 +1,46 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import AppInputWithIcon from "@/components/app-input-with-icon/AppInputWithIcon";
+
+describe("AppInputWithIcon ", () => {
+  test("Should render search input field", () => {
+    render(<AppInputWithIcon placeholder="Search..." />);
```

Comment:
try to write this in `beforeEach` block and not to write in each test

Reply:
It's redundent  redundent there, because each render hes different props

Reply:
> It's redundent redundent there, because each render hes different props

Mock your fns like `handleSearch` and `handleClear` pass in the `AppInputWithIcon` like mocked value
and define `beforeEach` block which will shorten the render execution only once in this block not reason to write it in each test

Reply:
Moved!